### PR TITLE
Switch TabControl tab if current tab is closed

### LINKED ISSUE #3152: Tab control should select next tab when current tab is closed

### PR DESCRIPTION
**Pull request by crabn3bula**
#3152

### CODE REVIEW

**Comment by smoogipoo** [comment]
Please add a test for this, `TestSceneTabControl` is a good place for it.

**Comment by smoogipoo** [comment]
Seems like CI didn't like your formatting. Please run `dotnet format` inside `TabControl.cs` and `TestSceneTabControl.cs`.

File: osu.Framework/Graphics/UserInterface/TabControl.cs [hunk 1]
```diff
@@ -231,6 +231,7 @@ protected virtual void AddTabItem(TabItem<T> tab, bool addToDropdown = true)
 
         /// <summary>
         /// Removes a <see cref="TabItem{T}"/> from this <see cref="TabControl{T}"/>.
+        /// If the tab to be removed is currently selected, then the next one is selected, otherwise the previous one is selected
```
[the remark this answers is not in the document]
Comment (crabn3bula) [reply osu.Framework/Graphics/UserInterface/TabControl.cs:234]: Whoops. Sorry, This is my first attempt to contribution.

File: osu.Framework/Graphics/UserInterface/TabControl.cs [hunk 2]
```diff
@@ -240,7 +242,15 @@ protected virtual void RemoveTabItem(TabItem<T> tab, bool removeFromDropdown = t
                 throw new InvalidOperationException($"Cannot remove non-removable tab {tab}. Ensure {nameof(TabItem.IsRemovable)} is set appropriately.");
 
             if (tab == SelectedTab)
-                SelectedTab = null;
+            {
+                if (Items.Count == 1)
+                    SelectedTab = null;
+                else
+                {
+                    tabMap.TryGetValue(Items.LastOrDefault(), out TabItem<T> lastTab);
```
Comment (bdach) [inline osu.Framework/Graphics/UserInterface/TabControl.cs:250]: I'm internally debating whether this shouldn't be
```suggestion
                    tabMap.TryGetValue(Items.Last(), out TabItem<T> lastTab);
```
Reasoning is:

* Right now rider complains (rightly) that `TryGetValue` could throw due to trying to pass a null key.
* `RemoveTabItem` should never get called if `Items.Count == 0`.

This should never fail, but I think dropping the `OrDefault` would make for a saner (and easier to debug) failure.

**Comment by crabn3bula** [comment]
I've applied the suggestions. Thank you for your patience and time spent, I will try not to make all these mistakes in the future.

**Comment by bdach** [comment]
I've rewritten the xmldoc slightly. Other than that I think this is alright.

File: osu.Framework/Graphics/UserInterface/TabControl.cs [hunk 3]
```diff
@@ -240,7 +250,15 @@ protected virtual void RemoveTabItem(TabItem<T> tab, bool removeFromDropdown = t
                 throw new InvalidOperationException($"Cannot remove non-removable tab {tab}. Ensure {nameof(TabItem.IsRemovable)} is set appropriately.");
 
             if (tab == SelectedTab)
-                SelectedTab = null;
+            {
+                if (Items.Count == 1)
+                    SelectedTab = null;
+                else
+                {
+                    tabMap.TryGetValue(Items.Last(), out TabItem<T> lastTab);
```
Comment (peppy) [inline osu.Framework/Graphics/UserInterface/TabControl.cs:258]: ugh this completely falls over with how `ChannelTabControl` is implemented

Comment (peppy) [reply osu.Framework/Graphics/UserInterface/TabControl.cs:258]: there's no safety to check if the same tab is selected, and the change is being performed before the tab is removed from the `Items` being read here.

**Pull request merged**